src/main/java/de/tum/cit/aet/artemis/iris/web/IrisResource.java

```diff
@@ -47,4 +74,66 @@ public ResponseEntity<IrisStatusDTO> getStatus() {
         return ResponseEntity.ok(new IrisStatusDTO(health.getStatus() == Status.UP, rateLimitInfo));
     }
 
+    /**
+     * Retrieves the overall ingestion state of a lecture by communicating with Pyris.
+     *
+     * <p>
+     * This method sends a GET request to the external Pyris service to fetch the current ingestion
+     * state of all lectures in a course, identified by its `lectureId`. The ingestion state can be aggregated from
+     * multiple lecture units or can reflect the overall status of the lecture ingestion process.
+     * </p>
+     *
+     * @param courseId the ID of the lecture for which the ingestion state is being requested
+     * @return a {@link ResponseEntity} containing the {@link IngestionState} of the lecture,
+     */
+    @GetMapping("courses/{courseId}/lectures/ingestion-state")
+    @EnforceAtLeastInstructorInCourse
+    public ResponseEntity<Map<Long, IngestionState>> getStatusOfLectureIngestion(@PathVariable long courseId) {
+        try {
+            Course course = courseRepository.findByIdElseThrow(courseId);
+            authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
```

Comment:
I believe this auth check is entirely redundant because you already have the annotation

Reply:
stephan wants to check that the user is an instructor in that specific course